### PR TITLE
[Ray] Implements get_total_n_cpu for Ray execution context

### PR DESCRIPTION
**Pull request by fyrestone**
## What do these changes do?

- [x] Implements `get_total_n_cpu` for Ray execution context.

## Related issue number

https://github.com/mars-project/mars/issues/2893

## Check code requirements

- [ ] tests added / passed (if needed)
- [ ] Ensure all linting tests pass, see [here](https://docs.pymars.org/en/latest/development/contributing.html#check-code-styles) for how to run them

### CODE REVIEW

File: mars/services/task/execution/ray/tests/test_ray_execution_backend.py [hunk 1]
```diff
@@ -195,7 +195,8 @@ def fake_init(self):
         pass
 
     with mock.patch.object(ThreadedServiceContext, "__init__", new=fake_init):
-        context = RayExecutionContext({"abc": o}, {}, None)
+        config = RayExecutionConfig.from_execution_config({"backend": "mars"})
```
Comment (chaokunyang) [inline mars/services/task/execution/ray/tests/test_ray_execution_backend.py:198]: Why passing `mars` to `RayExecutionConfig`

File: mars/services/task/execution/ray/context.py [hunk 2]
```diff
@@ -144,6 +153,11 @@ def get_chunks_meta(
             result.append(meta)
         return result
 
+    @implements(Context.get_total_n_cpu)
+    def get_total_n_cpu(self) -> int:
+        # TODO(fyrestone): Support auto scaling.
+        return self._config.get_n_cpu() * self._config.get_n_worker()
```
Comment (chaokunyang) [inline mars/services/task/execution/ray/context.py:159]: - What happed if `n_cpu` and `n_worker` not configured into ray config?
- Seems no case cover this

Comment (chaokunyang) [reply mars/services/task/execution/ray/context.py:159]: Why `mars` doesn't need to implement `get_total_n_cpu`?

File: mars/services/task/execution/ray/tests/test_ray_execution_backend.py (hunk 1 above)
Comment (fyrestone) [reply mars/services/task/execution/ray/tests/test_ray_execution_backend.py:198]: It's a mock config, I will fix it.

File: mars/services/task/execution/ray/context.py (hunk 2 above)
Comment (fyrestone) [reply mars/services/task/execution/ray/context.py:159]: > * What happed if `n_cpu` and `n_worker` not configured into ray config?
> * Seems no case cover this

The `n_cpu` and `n_worker` will be set when starting local cluster: https://github.com/mars-project/mars/blob/master/mars/deploy/oscar/local.py#L162. If these fields are not set, there should be a bug.

Mars backend has implemented the `get_total_n_cpu` API: https://github.com/mars-project/mars/blob/master/mars/services/context.py#L113

**Review by zhongchun — APPROVED** [review]
LGTM.

**Review by qinxuye — APPROVED** [review]
LGTM

**Pull request merged**